### PR TITLE
Updated JSON reader to throw error on missing file.

### PR DESCRIPTION
**Pull request by hoytak**

### CODE REVIEW

File: src/unity/lib/unity_sarray.cpp [hunk 1]
```diff
@@ -234,66 +234,80 @@ void unity_sarray::construct_from_json_record_files(std::string url) {
...
+          break;
         }
-      } else {
-        logstream(LOG_PROGRESS) << "Unable to read " << sanitize_url(p.first) << std::endl;
-      }
-    }
+      case fileio::file_status::DIRECTORY:
+        break; 
+      case fileio::file_status::MISSING:
+        log_and_throw("File '" + p.first + "' not found.");
```
Comment (znation) [inline src/unity/lib/unity_sarray.cpp]: Should these be `log_and_throw_io_failure`?

File: src/unity/lib/unity_sarray.cpp [hunk 2]
```diff
@@ -234,66 +234,80 @@ void unity_sarray::construct_from_json_record_files(std::string url) {
...
+                << "Unable to read " << sanitize_url(p.first) << std::endl;
+          }
+          break;
         }
-      } else {
-        logstream(LOG_PROGRESS) << "Unable to read " << sanitize_url(p.first) << std::endl;
-      }
-    }
+      case fileio::file_status::DIRECTORY:
+        break;
```
Comment (znation) [inline src/unity/lib/unity_sarray.cpp]: Should reading from a directory error out with a user-facing message? It seems this logic may result in an empty SArray/SFrame when attempting to read a directory, which is not quite what the user might expect.

Comment (hoytak) [reply src/unity/lib/unity_sarray.cpp]: Updated in PR

**Comment by hoytak** [comment]
Updated comments.

**Pull request merged**